mlfinlab/portfolio_optimization/hcaa.py

```diff
@@ -352,9 +355,11 @@ def allocate(self,
                  covariance_matrix=None,
                  expected_asset_returns=None,
                  allocation_metric='equal_weighting',
+                 linkage='average',
```

Comment:
Here you set the default to avg but in the other functions you have no defaults, should we add default linkage to the other functions?

Reply:
This is because users wont be able to call the other functions. The linkage in other methods takes values from this one. So just specifying one default value here works

Reply:
Thanks.